code/src/App.js

```diff
@@ -1,9 +1,115 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import moment from 'moment'
+import { API_URL } from './utils/urls'
+
 
 export const App = () => {
+
+
+  const [newThought, setNewThought] = useState([])
+  const [thoughts, setThoughts] = useState([])
+  const [characters, setCharacters] = useState([])
+
+
+  useEffect(() => {
+
+    fetch(API_URL)
+    .then((res) => res.json())
+    .then(data => setThoughts(data))
+
+  }, [] )
+
+
+  const onFormSubmit = (e) => {
+    e.preventDefault()
+    
+    fetch( API_URL, 
+        { method: 'POST', headers: { 'Content-Type': 'application/json',}, body: JSON.stringify({ message: newThought }) }
+      ).then((res) => res.json()).then((data) => setThoughts([data, ...thoughts])).then(setNewThought(''))
```

Comment:
To get the error message, in case the fetch is not successful. You can add `catch(error => console.log(error) `right after `.then`